### ISSUE TITLE
PixiJs 4.x -> 5.x, textures are shifted and breaks displaying

### ISSUE THREAD

**Issue by alcalyn** [DragonBones/DragonBonesJS#112]
I have a working animation in Pixi.js 4.8 + dragonbones.

I wanted to upgrade to pixi.js 5 and adapter DragonBones/Pixi/5.x.

This breaks all textures.

Broken, Pixi5.0 + DragonBones/Pixi/5.x:
![Pixi5.0 + DragonBones/Pixi/5.x](https://user-images.githubusercontent.com/1588144/61533761-16d84300-aa2e-11e9-8120-3696c5792856.png)

Expected, Pixi4.8 + DragonBones/Pixi/4.x:
![Pixi4.8 + DragonBones/Pixi/4.x](https://user-images.githubusercontent.com/1588144/61533887-6159bf80-aa2e-11e9-8b9a-1eb67c705fbf.png)

### Reproducing with jsFiddle

- Bugged, pixi.js 5, Dragonbones/Pixi/5.x: https://jsfiddle.net/xorpu2em/
- Working, pixi.js 4, Dragonbones/Pixi/4.x: https://jsfiddle.net/50x6qmpa/

Both jsfiddles use the same textures.

I tried to reproduce with the example from the official documentation: https://pixijs.io/examples/#/plugin-dragonbones/robot.js by animating it with pixijs 4, but it works for both pixi4 and 5:

- Pixi 5: https://jsfiddle.net/x36ukt48/
- Pixi 4: https://jsfiddle.net/x36ukt48/2/

I checked into diffs between DragonBones/Pixi/4.x / 5.x to see if I could find the difference, but didn't found anything to fix.

Here are my textures and animation data, Front_ske and Front_tex:
[Front.zip](https://github.com/DragonBones/DragonBonesJS/files/3411020/Archive.zip)

**Comment by alcalyn** [comment]
Here is a jsFiddle with the bug: https://jsfiddle.net/xorpu2em/

**Comment by JetLua** [comment]
Maybe @rtalon83  can help you.

**Comment by JetLua** [comment]
@ivanpopelyshev Do you have time to check it out? Maybe it has to do with `mesh`?

**Comment by ivanpopelyshev** [comment]
Maybe. Mesh textures are different in v5, need to adjust plugins for it. I'll try to make PR today or tomorrow.

**Comment by JetLua** [comment]
Fixed: https://jsfiddle.net/JetLu/g6854zbh/

```js
if (currentTextureData.rotated) {
  uvs[i] = 1 - v
  uvs[i + 1] = u
} else {
  uvs[i] = u
  uvs[i + 1] = v
}
```

Replace the code below:

https://github.com/DragonBones/DragonBonesJS/blob/be28ef831bfd91bd1f1353352a7edd946de01c54/Pixi/5.x/src/dragonBones/pixi/PixiSlot.ts#L219-L230

**Comment by ivanpopelyshev** [comment]
yeah, rotations aren't needed in v5 because pixi does it automagically.

**Comment by ivanpopelyshev** [comment]
@JetLua im not an expert on that DragonBones code, but it looks like Texture-related code can be simplified seriously.

Here is how it works in spine: https://github.com/pixijs/pixi-spine/blob/master/src/core/Texture.ts#L88

and that's all. If you assign texture with frame/orig/trim to mesh or sprite, pixi will take into account automatically.

**Comment by JetLua** [comment]
I'm a novice, too. 

There are indeed many things to optimize.

**Comment by rtalon83** [comment]
For me, dragonbones is outdated long time, but people continue using It
because is free. I havent time to improve db code, my firts contribution
here was very fast and run the firts time.

**Comment by alcalyn** [comment]
Tried the version from this PR: https://github.com/DragonBones/DragonBonesJS/pull/114 and it works!

> For me, dragonbones is outdated long time, but people continue using It because is free.

Actually I'm not using DragonBones Pro software, but just the json format (generated by a Blender plugin) and this Javascript Pixi plugin.

I found that DragonBones was the best solution to create skeleton. Its json format seems to be the most open (compared to Spine) and used, thanks to the work that has be done i.e in Godot or coa_tools blender plugin (https://github.com/ndee85/coa_tools/issues/26).
